### PR TITLE
GCC Warnings Fixed

### PR DESCRIPTION
**Pull request by luav**
Fix for  #1267

### CODE REVIEW

File: include/igraph_error.h [hunk 1]
```diff
@@ -563,7 +563,11 @@ DECLDIR int IGRAPH_FINALLY_STACK_SIZE(void);
 #define IGRAPH_FINALLY(func,ptr) \
   IGRAPH_FINALLY_REAL((igraph_finally_func_t*)(func), (ptr))
 
-#if (defined(__GNUC__) && GCC_VERSION_MAJOR >= 3)
+#if !defined(GCC_VERSION_MAJOR) && defined(__GNUC__)
+#  define GCC_VERSION_MAJOR  __GNUC__
```
Comment (ntamas) [inline include/igraph_error.h]: Not sure whether this is correct.  AFAIK `__GNUC__` is just a "marker" that is specified when we are using the GNU C compiler. `GCC_VERSION_MAJOR` contains the major version of the GCC compiler. Simply defining `GCC_VERSION_MAJOR` as `__GNUC__` seems fishy.

Which compiler were you using that defined `__GNUC__` but not `GCC_VERSION_MAJOR`?

**Comment by ntamas** [comment]
`igraph_sample_sphere_volume()` is defined in `src/dotproduct.c` and it can be used in conjunction with `igraph_dot_product_game()` to generate random graphs. Why is it removed?

File: include/igraph_error.h (hunk 1 above)
Comment (luav) [reply include/igraph_error.h]: "GCC version x.y.z defines `__GNUC__` to x, `__GNUC_MINOR__` to y, and `__GNUC_PATCHLEVEL__` to z.", see the [official GCC docs](https://gcc.gnu.org/onlinedocs/cpp/Common-Predefined-Macros.html).
I used `gcc version 5.4.0 20160609 (Ubuntu 5.4.0-6ubuntu1~16.04.12)` on Ubuntu 16.04 x64.

**Comment by luav** [comment]
@ntamas `igraph_sample_sphere_volume()` had **subsequent** *duplicated declarations in the same file* (`include/igraph_games.h`):
```c
DECLDIR int igraph_sample_sphere_surface(igraph_integer_t dim, igraph_integer_t n,
                igraph_real_t radius, 
                igraph_bool_t positive, 
                igraph_matrix_t *res);

/* DECLDIR int igraph_sample_sphere_volume(igraph_integer_t dim, igraph_integer_t n,
                igraph_real_t radius,
                igraph_bool_t positive,
                igraph_matrix_t *res);
/* ln. 217, declaration duplicating the following one is removed */

/* This declaration is remained */
DECLDIR int igraph_sample_sphere_volume(igraph_integer_t dim, igraph_integer_t n,
                igraph_real_t radius,
                igraph_bool_t positive,
                igraph_matrix_t *res);

DECLDIR int igraph_sample_dirichlet(igraph_integer_t n, const igraph_vector_t *alpha,
                igraph_matrix_t *res);
```

`igraph_rng_get_exp()` had *duplicated declarations in the same file* (`include/igraph_random.h`):
```c
DECLDIR igraph_real_t igraph_rng_get_exp(igraph_rng_t *rng, igraph_real_t rate);  /* 1st declaraion */
DECLDIR unsigned long int igraph_rng_get_int31(igraph_rng_t *rng);
/* DECLDIR igraph_real_t igraph_rng_get_exp(igraph_rng_t *rng, igraph_real_t rate);  /* ln.84, duplicated declaration is removed */
DECLDIR igraph_real_t igraph_rng_get_gamma(igraph_rng_t *rng, igraph_real_t shape,
```

`igraph_transitive_closure_dag()` is remained in the `include/igraph_topology.h` together with other DAG processing API:
```c
/* -------------------------------------------------- */
/* Directed acyclic graphs                            */
/* -------------------------------------------------- */

DECLDIR int igraph_topological_sorting(const igraph_t *graph, igraph_vector_t *res,
			       igraph_neimode_t mode);
DECLDIR int igraph_is_dag(const igraph_t *graph, igraph_bool_t *res);
DECLDIR int igraph_transitive_closure_dag(const igraph_t *graph,
				  igraph_t *closure);
```

File: include/igraph_error.h (hunk 1 above)
Comment (ntamas) [reply include/igraph_error.h]: Hmmm, indeed. How on Earth could this have worked before?

**Comment by ntamas** [comment]
Thanks -- sorry, I missed that in the diff somehow! It is merged now.

**Pull request merged**